client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -135,22 +143,55 @@ private boolean isMemoryShuffleEnabled(String storageType) {
   public void write(Iterator<Product2<K, V>> records) throws IOException {
     List<ShuffleBlockInfo> shuffleBlockInfos = null;
     Set<Long> blockIds = Sets.newConcurrentHashSet();
+
+    boolean isCombine = shuffleDependency.mapSideCombine();
+    SizeTrackingAppendOnlyMap<K, V> sizeTrackingAppendOnlyMap = null;
+    if (isCombine) {
+      sizeTrackingAppendOnlyMap = new SizeTrackingAppendOnlyMap<>();
+    }
+
```

Comment:
`SizeTrackingAppendOnlyMap<K, V> sizeTrackingAppendOnlyMap = new SizeTrackingAppendOnlyMap<>();` will be more clean.